packages/nextjs/src/middleware/helpers.ts

```diff
@@ -57,3 +57,65 @@ export const getHostedLogoutUrl = (referer = config.appUrl): BuildRouteResult =>
 
   return buildLogoutRoute(redirectUrl, config.baseUrl);
 };
+
+export type Tokens = {
+  accessToken: string;
+  refreshToken: string;
+};
+
+/**
+ * Extracts the access token from the response body
+ * @param bodyStr
+ */
+export const extractAccessToken = (bodyStr: string): Tokens => {
+  const body = JSON.parse(bodyStr);
+
+  if (body.authResponse) {
+    Object.assign(body, body.authResponse);
+  }
+  return {
+    accessToken: body.accessToken || body.access_token,
+    refreshToken: body.refreshToken || body.refresh_token,
+  };
+};
+
+const jwtKeys = ['accessToken', 'access_token', 'idToken', 'id_token'];
```

Comment:
Just note that in some cases the accessToken is not a JWT

Reply:
access token in implicit mode is still a JWT